src/wp-includes/media.php

```diff
@@ -5468,53 +5542,79 @@ function wp_get_webp_info( $filename ) {
- *                     that the `loading` attribute should be skipped.
+ * @param string $tag_name The tag name.
+ * @param array  $attr     Array of the attributes for the tag.
+ * @param string $context  Context for the element for which the loading optimization attribute is requested.
+ * @return array Loading optimization attributes.
  */
-function wp_get_loading_attr_default( $context ) {
+function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
 	global $wp_query;
 
-	// Skip lazy-loading for the overall block template, as it is handled more granularly.
+	$loading_attrs = array();
+
+	/*
+	 * Skip lazy-loading for the overall block template, as it is handled more granularly.
+	 * The skip is also applicable for `fetchpriority`.
+	 */
 	if ( 'template' === $context ) {
-		return false;
+		return $loading_attrs;
 	}
 
-	// Do not lazy-load images in the header block template part, as they are likely above the fold.
-	// For classic themes, this is handled in the condition below using the 'get_header' action.
+	// For now this function only supports images and iframes.
+	if ( 'img' !== $tag_name && 'iframe' !== $tag_name ) {
+		return $loading_attrs;
+	}
+
+	// For any resources, width and height must be provided, to avoid layout shifts.
+	if ( ! isset( $attr['width'] ) || ! isset( $attr['height'] ) ) {
+		return $loading_attrs;
+	}
+
+	if ( isset( $attr['loading'] ) ) {
+		if ( 'lazy' === $attr['loading'] ) {
+			$loading_attrs['loading'] = $attr['loading'];
+			if ( isset( $attr['fetchpriority'] ) && 'high' === $attr['fetchpriority'] ) {
+				_doing_it_wrong(
+					__FUNCTION__,
+					__( 'An image should not be lazy-loaded and marked as high priority at the same time.' ),
+					'6.3.0'
+				);
+			}
+		}
+
+		return wp_maybe_strip_loading_attribute( wp_maybe_add_fetchpriority_high_attr( $loading_attrs, $tag_name, $attr ), $tag_name, $context );
+	}
+
+	// An image with `fetchpriority="high"` cannot be assigned `loading="lazy"` at the same time.
+	if ( isset( $attr['fetchpriority'] ) && 'high' === $attr['fetchpriority'] ) {
+		return wp_maybe_strip_loading_attribute( wp_maybe_add_fetchpriority_high_attr( $loading_attrs, $tag_name, $attr ), $tag_name, $context );
+	}
+
+	/*
+	 * Do not lazy-load images in the header block template part, as they are likely above the fold.
+	 * For classic themes, this is handled in the condition below using the 'get_header' action.
+	 */
 	$header_area = WP_TEMPLATE_PART_AREA_HEADER;
 	if ( "template_part_{$header_area}" === $context ) {
-		return false;
+		return wp_maybe_strip_loading_attribute( wp_maybe_add_fetchpriority_high_attr( $loading_attrs, $tag_name, $attr ), $tag_name, $context );
 	}
 
 	// Special handling for programmatically created image tags.
-	if ( ( 'the_post_thumbnail' === $context || 'wp_get_attachment_image' === $context ) ) {
+	if ( 'the_post_thumbnail' === $context || 'wp_get_attachment_image' === $context ) {
```

Comment:
Why is this change again?

Reply:
@spacedmonkey 
I feel git is a bit confused here after we sifted old function to deprecated.php. Both old and new function had similar lines of code so git is trying to match similar lines. 

Now for the following change, it was modified based on earlier review. 
 https://github.com/WordPress/wordpress-develop/pull/4495#discussion_r1213467091

I have made the same change in the older function too reasoning - 
https://github.com/WordPress/wordpress-develop/pull/4495#discussion_r1231135533